packages/box/src/Box.spec.tsx

```diff
@@ -38,89 +24,109 @@ describe('packages/box', () => {
     });
 
     test('it renders the child content', () => {
-      expect(renderedEls.child).toBeInTheDocument();
+      expect(child).toBeInTheDocument();
     });
 
     test('it preserves the shared props', () => {
-      expect(renderedEls.box).toHaveAttribute('name', sharedProps.name);
+      expect(box).toHaveAttribute('name', sharedProps.name);
     });
   });
 
   describe('when rendered with the expected anchor props', () => {
+    let box: HTMLElement;
+    let child: HTMLElement;
+
     beforeEach(() => {
-      renderBox(anchorProps);
+      const { getByTestId } = render(
+        <Box {...anchorProps} data-testid="anchor-component">
+          <span data-testid="anchor-child">hi</span>
+        </Box>,
+      );
+
+      box = getByTestId('anchor-component');
+      child = getByTestId('anchor-child');
     });
 
     test('it renders the box component as an anchor', () => {
-      expect(renderedEls.box).toBeInTheDocument();
-      expect(renderedEls.box?.tagName.toLowerCase()).toBe('a');
+      expect(box).toBeInTheDocument();
+      expect(box?.tagName.toLowerCase()).toBe('a');
     });
 
     test('it renders the child content', () => {
-      expect(renderedEls.child).toBeInTheDocument();
+      expect(child).toBeInTheDocument();
     });
 
     test('it preserves the shared props', () => {
-      expect(renderedEls.box).toHaveAttribute('name', sharedProps.name);
+      expect(box).toHaveAttribute('name', sharedProps.name);
     });
 
     test('it sets the anchor-specific attributes', () => {
-      expect(renderedEls.box).toHaveAttribute('href', anchorProps.href);
-      expect(renderedEls.box).toHaveAttribute('target', anchorProps.target);
+      expect(box).toHaveAttribute('href', anchorProps.href);
+      expect(box).toHaveAttribute('target', anchorProps.target);
     });
   });
 
   describe('when rendered as a custom component', () => {
+    let box: HTMLElement;
+    let child: HTMLElement;
+
     beforeEach(() => {
-      renderBox(linkProps);
+      const { getByTestId } = render(
+        <Box
+          as={LinkWrapper}
+          href="https://cloud.mongodb.com"
+          target="_blank"
+          data-testid="custom-component"
+        />,
+      );
+      box = getByTestId('custom-component');
+      child = getByTestId('link-component');
     });
 
     test('it renders the box component as the custom component', () => {
-      expect(renderedEls.box).toBeInTheDocument();
+      expect(box).toBeInTheDocument();
       expect(linkWrapperFn).toHaveBeenCalledTimes(1);
-      expect(renderedEls.box?.tagName.toLowerCase()).toBe('span');
+      expect(box?.tagName.toLowerCase()).toBe('span');
     });
 
     test('it renders the child content', () => {
-      expect(renderedEls.child).toBeInTheDocument();
-    });
-
-    test('it preserves the shared props', () => {
-      expect(renderedEls.box).toHaveAttribute('name', sharedProps.name);
+      expect(child).toBeInTheDocument();
     });
   });
 });
```

Comment:
could we have a TODO to add "type tests" that test that type errors occur if there are missing or invalid props? we should be able to write these tests after we upgrade TS.